src/rgw/rgw_acl.h

```diff
@@ -214,57 +217,30 @@ struct ACLReferer {
       perm(perm) {
   }
 
-  bool is_match(std::string http_referer) const {
-    std::string http_host;
-    if (!get_http_host(http_referer, http_host))
+  bool is_match(boost::string_ref http_referer) const {
+    boost::optional<boost::string_ref> http_host = get_http_host(http_referer);
+    if ((*http_host).empty()) {
```

Comment:
In `get_http_host` you might want to return `boost::none` if parsing fails. Then this conditional would look like:
```
if (! http_host) {
  return false;
}
```